app/transport/rest/user.go

```diff
@@ -106,3 +106,47 @@ func (uh UserHandler) GetUserByID() http.Handler {
 		WriteJSONResponse(w, http.StatusOK, user, uh.logger)
 	})
 }
+
+// DeleteUser will handle user creation.
+func (uh UserHandler) DeleteUser() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		var user entities.User
+
+		if err := json.NewDecoder(req.Body).Decode(&user); err != nil {
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+			uh.logger.Errorf("Failed decoding JSON from request %+v: %+v", req, err)
+
+			return
+		}
+
+		defer func() {
+			if err := req.Body.Close(); err != nil {
+				uh.logger.Warnf("Failed closing request %+v: %+v", req, err)
+			}
+		}()
+
+		if err := user.ValidateUUID(); err != nil {
+			uh.logger.Warnf("Invalid ID in the request: %s", user)
```

Comment:
Good point. It's probably worth adopting the same style for the entire project. It would be nice to fix this throughout the project code. Therefore, I wrote it down as a separate task